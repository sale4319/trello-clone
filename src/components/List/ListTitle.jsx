import React, { useEffect, useState } from 'react';
import { RiDeleteBin5Line } from 'react-icons/ri';
import { useCloseList, useEditList } from '../../api/apiHooks/apiLists';

const ListTitle = ({ name, listId }) => {
    const [deleteListId, setDeleteListId] = useState('');
    useCloseList(deleteListId);

    const [isEditingName, setEditingName] = useState(false);
    const [formName, setFormName] = useState(name);
    const [submitNewFormName, setSubmitNewFormName] = useState('');
    useEditList(listId, submitNewFormName);

    useEffect(() => {
        return () => {
            setSubmitNewFormName('');
            setDeleteListId('');
        };
    }, []);

    const handleCloseList = () => {
        setDeleteListId(listId);
        return;
    };

    const handleEnterKeyPress = (event) => {
        if (event.key === 'Enter' && name !== formName && !!formName) {
            setSubmitNewFormName(formName);
            setEditingName(false);
        }
    };

    const handleEditComplete = () => {
        if (name !== formName && !!formName) {
            setSubmitNewFormName(formName);
        }
        setEditingName(false);
    };

    return (
        <div className="list-title-container">
            {isEditingName ? (
                <input
                    autoFocus
                    value={formName}
                    onChange={event => setFormName(event.target.value)}
                    onKeyPress={handleEnterKeyPress}
                    onBlur={handleEditComplete}
                />
            ) : (
                <div className="list-title-title" onClick={() => setEditingName(true)}>
                    {name}
                </div>
            )}
            <div className="list-title-icon-container" onClick={handleCloseList}>
                <RiDeleteBin5Line size="30" />
            </div>
        </div>
    );
};

export default ListTitle;