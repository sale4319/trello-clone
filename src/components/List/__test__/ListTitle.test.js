import React from 'react';
import ListTitle from '../ListTitle';

import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

test('component renders', () => {
    renderer.create(<ListTitle />).getInstance();

  });

test("mateches snapshot", () => {
    const tree = renderer.create(<ListTitle/>).toJSON();
    expect(tree).toMatchSnapshot();
});