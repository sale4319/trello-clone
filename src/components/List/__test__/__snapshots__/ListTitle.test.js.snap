// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mateches snapshot 1`] = `
<div
  className="list-title-container"
>
  <div
    className="list-title-title"
    onClick={[Function]}
  />
  <div
    className="list-title-icon-container"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="30"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M4 8h16v13a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V8zm2 2v10h12V10H6zm3 2h2v6H9v-6zm4 0h2v6h-2v-6zM7 5V3a1 1 0 0 1 1-1h8a1 1 0 0 1 1 1v2h5v2H2V5h5zm2-1v1h6V4H9z"
        />
      </g>
    </svg>
  </div>
</div>
`;
